mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_WebViewPopup.java

```diff
@@ -145,9 +152,40 @@ public void onDismiss(DialogInterface dialog) {
 
     private void openDialogPopup() {
         if (mDialog.isShowing()) return;
+
+        // Set the dialog content view to match parent's height and width
+        ViewGroup.LayoutParams params = new ViewGroup.LayoutParams(
+            ViewGroup.LayoutParams.MATCH_PARENT,
+            ViewGroup.LayoutParams.MATCH_PARENT
+        );
+        mDialog.getWindow().setContentView(this, params);
+
+        // Set horizontal and vertical margins
+        WindowManager.LayoutParams layoutParams = new WindowManager.LayoutParams();
+        layoutParams.copyFrom(mDialog.getWindow().getAttributes());
+
+        // Set width and height to match_parent
+        layoutParams.width = WindowManager.LayoutParams.MATCH_PARENT;
+        layoutParams.height = WindowManager.LayoutParams.MATCH_PARENT;
+
+        // Calculate the margin size
```

Comment:
This is what we should get rid of.

That's what I meant by the last comment.

Set width/height with constants
Set margins as percentage
Do not calculate margins on our own, let the framework do that thanks to the widths and margins configured.

That's not possible? 🤷

Reply:
No, I don't know how to do that